src/types.ts

```diff
@@ -65,58 +76,68 @@ export interface PagerViewProps {
-   *   - 'on-drag', the keyboard is dismissed when a drag begins.
+   * Orientation of pager.
    */
-  keyboardDismissMode?: 'none' | 'on-drag';
+  orientation?: Orientation;
 
   /**
-   * Blank space to show between pages. This is only visible while scrolling, pages are still
-   * edge-to-edge.
+   * Determines whether it's possible to overscroll a bit
+   * after reaching end or very beginning of pages.
+   */
+  overdrag?: boolean;
+
+  /**
+   * Blank space to show between pages. This is only visible while scrolling,
+   * pages are still edge-to-edge.
    */
   pageMargin?: number;
 
+  /**
+   * When false, the content does not scroll.
+   * The default value is true.
+   */
+  scrollEnabled?: boolean;
+
+  /**
+   * Shows the dots indicator at the bottom of the view.
+   */
+  showPageIndicator?: boolean;
+
   style?: ReactNative.StyleProp<ReactNative.ViewStyle>;
+
   /**
-   * Set the number of pages that should be retained to either side
-   * of the currently visible page(s). Pages beyond this limit will
-   * be recreated from the adapter when needed.
-   * Defaults to RecyclerView's caching strategy.
-   * The given value must either be larger than 0.
-   *
-   * Only supported on Android.
+   * Transition style of pager.
    */
-  offscreenPageLimit?: number;
+  transitionStyle?: TransitionStyle;
+}
 
-  children: ReactNode;
+export interface PagerViewProps<ItemT> extends EagerPagerProps {
+  /**
+   * Number of pages to render before/after the current page. Minimum 1.
+   */
+  buffer?: number;
 
   /**
-   * If a parent `View` wants to prevent a child `View` from becoming responder
-   * on a move, it should have this handler which returns `true`.
-   *
-   * `View.props.onMoveShouldSetResponderCapture: (event) => [true | false]`,
-   * where `event` is a synthetic touch event as described above.
-   *
-   * See http://facebook.github.io/react-native/docs/view.html#onMoveShouldsetrespondercapture
+   * Array of data to be rendered as pages.
    */
-  onMoveShouldSetResponderCapture?: (
-    event: ReactNative.GestureResponderEvent
-  ) => boolean;
+  data: ItemT[];
 
   /**
-   * iOS only
+   * Compute a unique key for the given item at the specified index.
    */
-  orientation?: Orientation;
-  transitionStyle?: TransitionStyle;
-  showPageIndicator?: boolean;
+  keyExtractor: (item: ItemT, index: number) => string;
+
   /**
-   * Android only
+   * Maximum number of pages allowed to stay rendered. Set to 0 for unlimited.
+   *
+   * Default unlimited. Will always render at least `1 + 2 * buffer` pages.
    */
-  overScrollMode?: OverScrollMode;
+  maxRenderWindow?: number;
+
   /**
-   * Determines whether it's possible to overscroll a bit
-   * after reaching end or very beginning of pages.
+   * Render an item from `data` into a page.
    */
-  overdrag?: boolean;
+  renderItem: (info: { item: ItemT; index: number }) => React.ReactElement;
 }
+
+export type PagerViewState = { offset: number; windowLength: number };
```

Comment:
Why this type is needed ?

Reply:
It is used to keep track, js side, of which views to render for native side.
- https://github.com/alpha0010/react-native-viewpager/blob/a13b821be982b686e919fe1db836304cd3df6800/src/PagerView.tsx#L147-L155
- https://github.com/alpha0010/react-native-viewpager/blob/a13b821be982b686e919fe1db836304cd3df6800/src/PagerView.tsx#L179